src/directives/onLongPress.ts

```diff
@@ -0,0 +1,137 @@
+import {ElementPart, nothing} from 'lit';
+import {AsyncDirective, directive, PartType} from 'lit/async-directive.js';
+import type {
+  DirectiveParameters,
+  DirectiveClass,
+  DirectiveResult,
+  PartInfo
+} from 'lit/async-directive.js';
+
+type LongPressCallback = (event?: PointerEvent) => void;
+
+const DEFAULT_LONG_PRESS_TIMEOUT_MS = 1000;
+
+class LongPressDirective extends AsyncDirective {
+  /** Element of the directive. */
+  #element!: Element;
+
+  /**
+   * Long press timeout.
+   * This timeout initiates when the pointer is pressed on the
+   * element. It calls user's callback unless cancel events occur
+   * before time's out.
+   */
+  #longPressTimeout?: NodeJS.Timeout;
```

Comment:
gah this is `@types/node` slipping into our types as per usual. in a separate pr i need to set `types` in tsconfig so this doesn't happen

this should actually be a `number`, not a `NodeJS.Timeout`

Reply:
Problem is TypeScript complains about `setTimeout` not returning a number, so I did this naughty casting trick..

Reply:
https://github.com/43081j/relit/blob/3d4418ebeee8178dc236f8f4642f36d7faad854b/src/directives/onLongPress.ts#L105

Reply:
i've sorted it in #20, can you catch up from `main`, then remove this cast and see if all works as expected

Reply:
Works great!